packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -98,9 +120,9 @@ export interface FieldDiscrepancyLocation {
 export interface AllowedTypeDiscrepancy extends FieldDiscrepancyLocation {
 	mismatch: "allowedTypes";
 	/**
-	 * List of allowed type identifiers in viewed schema which are not allowed in stored schema
+	 * List of annotated allowed types in viewed schema which are not allowed in stored schema
 	 */
-	view: TreeNodeSchemaIdentifier[];
+	view: readonly AnnotatedAllowedType<TreeNodeSchema>[];
```

Comment:
its seems like this interface should probably be fully read-only, or fully mutable depend on its use case.

If you want to make AllowedTypeDiscrepancy actually readonly, FieldDiscrepancyLocation would want to readonly its fields, then readonly this field, and the field and array below.

Reply:
Note that if readonly is the desired outcome, its fine to not fix all the other ones in this PR, and just leave this as you have it to reduce unnecessary changes in this larger PR.

Reply:
I removed the readonly to keep the change minimal, we can revisit whether this should be fully readonly in the follow up task.

Reply:
`AllowedTypeDiscrepancy.mismatch`, `FieldDiscrepancyLocation.identifier` and `FieldDiscrepancyLocation.fieldKey` are still mutable. Given everything else is readonly, it seems like missing those might be a mistake. If not might be good to leave a TODO for them.